Base/BusyThreads.cpp

```diff
@@ -68,24 +99,44 @@ void BusyThreads::teardown()
 
 void BusyThreads::busyThread(const int threadIndex)
 {
-  using Clock = std::chrono::high_resolution_clock;
+  // A busy thread alternates between blocking on a condition variable and performing
+  // low-energy work via a hardware delay instruction. Blocking avoids being terminated
+  // when running in the background by violating the iOS CPU usage limit. The CPU usage
+  // percentage needs to be set high enough to prevent CPU throttling and low enough to
+  // avoid background termination.
+  //
+  // The use of mIsActive and the condition variable allows busy threads to be quickly
+  // torn down when setNumThreads() is called or the class is destroyed.
```

Comment:
Does the period ever get long enough that it's worth the hassle?

Reply:
The Period slider maxes out at 100ms. When set to 100ms, the Threads slider stutters without this, since threads need to be joined while dragging. I'm also afraid of people copying this code and accidentally introducing a long delay when destroying the busy thread (e.g., upon document loading).